kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerObservableManualCommit.scala

```diff
@@ -47,12 +50,44 @@ final class KafkaConsumerObservableManualCommit[K, V] private[kafka] (
       Task(blocking(consumer.synchronized(consumer.commitSync(batch.map {
         case (k, v) => k -> new OffsetAndMetadata(v)
       }.asJava))))
-    override def commitBatchAsync(batch: Map[TopicPartition, Long], callback: OffsetCommitCallback): Task[Unit] =
-      Task {
-        blocking(consumer.synchronized(consumer.commitAsync(batch.map {
-          case (k, v) => k -> new OffsetAndMetadata(v)
-        }.asJava, callback)))
-      }
+    override def commitBatchAsync(batch: Map[TopicPartition, Long]): Task[Unit] = {
+      Task
+        .async0[Unit] { (s, cb) =>
+          val asyncCb = Callback.forked(cb)(s)
+          s.executeAsync { () =>
+            consumer.synchronized {
+              val isActive = Atomic(true)
+              try {
+                val offsets = batch.map {
+                  case (k, v) => k -> new OffsetAndMetadata(v)
+                }.asJava
+                consumer.commitAsync(
+                  offsets,
+                  new OffsetCommitCallback {
```

Comment:
I think we should be able to reuse reference to this `Callback` when we upgrade to Monix 3.0.0 which has relaxed contract and we might not need `isActive` check

Reply:
Fixed this, new callback API is much more convenient